project/src/main/ui/keybind/settings-keybinds-control.gd

```diff
@@ -0,0 +1,64 @@
+extends VBoxContainer
+"""
+A UI control which lets the player view and update the game's keybinds.
+"""
+
+func _ready() -> void:
+	$Presets/Guideline.connect("pressed", self, "_on_Guideline_pressed")
+	$Presets/Wasd.connect("pressed", self, "_on_Wasd_pressed")
+	$Presets/Custom.connect("pressed", self, "_on_Custom_pressed")
+	$CustomScrollContainer/VBoxContainer/ResetToDefault.connect("pressed", self, "_on_ResetToDefault_pressed")
+	
+	match PlayerData.control_settings.preset:
+		ControlSettings.GUIDELINE: $Presets/Guideline.pressed = true
+		ControlSettings.WASD: $Presets/Wasd.pressed = true
+		ControlSettings.CUSTOM: $Presets/Custom.pressed = true
+	
+	_refresh_control_labels()
+
+
+"""
+Updates the UI controls to reflect the currently selected preset.
+
+The 'WASD' and 'Guideline' presets reuse the same UI controls and change the label contents.
+
+The 'Custom' preset swaps in different UI controls because it's interactive.
+"""
+func _refresh_control_labels() -> void:
+	$PresetScrollContainer.visible = false
+	$CustomScrollContainer.visible = false
+	
+	match PlayerData.control_settings.preset:
+		ControlSettings.GUIDELINE:
+			$PresetScrollContainer.visible = true
+			$PresetScrollContainer/VBoxContainer/MovePiece.value = "Left, Right, Numpad4/6"
+			$PresetScrollContainer/VBoxContainer/SoftDrop.value = "Down, Numpad2"
+			$PresetScrollContainer/VBoxContainer/HardDrop.value = "Space, Up, Shift, Numpad8"
+			$PresetScrollContainer/VBoxContainer/RotatePiece.value = "Z, X, Numpad7/9"
+		ControlSettings.WASD:
+			$PresetScrollContainer.visible = true
+			$PresetScrollContainer/VBoxContainer/MovePiece.value = "A, D, Numpad4/6"
+			$PresetScrollContainer/VBoxContainer/SoftDrop.value = "S, Down, Numpad8"
+			$PresetScrollContainer/VBoxContainer/HardDrop.value = "W, Up, Space, Numpad5"
```

Comment:
I suspect WASD players use the arrows for hard/soft drop. I don't think'd use S and W.

Reply:
I've reordered the controls and label to place Down/Up at the front. I verified it works in the tutorial as well.